### PR TITLE
Brapi v1: studies/id/germplasm call

### PR DESCRIPTION
**Pull request by CarlosHorro**
Official documentation:
https://github.com/plantbreeding/API/blob/master/Specification/Studies/StudyGermplasmDetails.md

Following structure and improvements of previous calls.

### CODE REVIEW

File: app/controllers/brapi/v1/studies_controller.rb [hunk 1]
```diff
@@ -121,6 +124,68 @@ def show
...
+      
+      if result_object.count == 0
+        render json: { reason: 'Resource not found' }, status: :not_found  # 404
+      else
+        json_result_array = []
+        
+        entry_number = 1 + ((page-1) * page_size)
+        # any programmatic data manipulation can be done here
+        result_object.each do |row|
+          row[:entryNumber] = entry_number.to_s
```
[the remark this answers is not in the document]
Comment (CarlosHorro) [reply app/controllers/brapi/v1/studies_controller.rb]: In the spec for this call:
https://github.com/plantbreeding/API/blob/master/Specification/Studies/StudyGermplasmDetails.md
There are no details of any return field, but you can see among them 'entryNumber' being used like a counter... I know that depending on the order the same entry can be given a different number, the name of the field suggested me that...but I could only guess its working...

File: app/controllers/brapi/v1/studies_controller.rb [hunk 2]
```diff
@@ -121,6 +124,68 @@ def show
...
+        # pagination data returned
+        
+        result_count_object = studies_queries.germplasm_query(query_params, count_mode: true )
+        
+        total_count = result_count_object.values.first[0].to_i
+        total_pages = (total_count/page_size.to_f).ceil
+        
+        json_response = { 
+          metadata: json_metadata(page_size, page, total_count, total_pages),
+          result: {
```
[the remark this answers is not in the document]
Comment (CarlosHorro) [reply app/controllers/brapi/v1/studies_controller.rb]: Um... I was not sure about this point, but thinking about it I think I can apply the mapping 
trials trialName -> plant_trials.project_descriptor

I will add it.

File: app/controllers/brapi/v1/studies_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kammerer) [reply app/controllers/brapi/v1/studies_controller.rb]: Well, if there is no more info anywhere then I suppose it fine as it is.

File: app/services/brapi/v1/studies_queries.rb [hunk 3]
```diff
@@ -113,7 +113,7 @@ def studies_search_query(query_params, count_mode:)
     else
       # order
       order_query =  " ORDER BY "+ get_sortby_field(sort_by)      
-      order_query += (sort_order != nil && sort_order == "desc"?" desc ":" asc ") 
+      order_query += (sort_order!=nil && sort_order=="desc"?" desc ":" asc ")
```
[the remark this answers is not in the document]
Comment (CarlosHorro) [reply app/services/brapi/v1/studies_queries.rb]: Yes, I think I had some problem with my local merge and I included some old stuff, sorry...

**Comment by CarlosHorro** [comment]
It seems that there is some problem checking specs, I'm looking into it.

**Comment by CarlosHorro** [comment]
studies/id/germplasm spec is working now

**Comment by CarlosHorro** [comment]
Is there everything OK to merge this branch?

**Comment by kammerer** [comment]
I don't know if you need `plant_population_lists` in your query but they are part of PP submission form so I am not sure if you can say they are no longer maintained.

**Comment by CarlosHorro** [comment]
Sorry, I don't want to mix this all with a new BrAPI call. I will create a new branch and pull request for that call.

**Comment by wjurkowski** [comment]
dependent on solving the PlantPopulationList topic

**Comment by CarlosHorro** [comment]
After talking with Wiktor, we've agreed that the best way to do this is using the path:
PlantTrial - PlantScoringUnit – PlantAccession
So I've changed the query to reflect it.
I think everything is now ready for merging this branch.

**Pull request merged**